core/src/main/java/com/google/instrumentation/stats/DistributionAggregation.java

```diff
@@ -29,64 +31,61 @@
  * <p>Although not forbidden, it is generally a bad idea to include non-finite values (infinities
  * or NaNs) in the population of values, as this will render the {@code mean} meaningless.
  */
-public final class DistributionAggregation {
+@Immutable
+@AutoValue
+public abstract class DistributionAggregation {
   /**
    * Constructs a new {@link DistributionAggregation}.
    */
-  public static final DistributionAggregation create(
+  public static DistributionAggregation create(
       long count, double mean, double sum, Range range, List<Tag> tags) {
-    return new DistributionAggregation(count, mean, sum, range, tags, null);
+    return createInternal(count, mean, sum, range, tags, null);
   }
 
   /**
    * Constructs a new {@link DistributionAggregation} with the optional {@code bucketCount}s.
    */
-  public static final DistributionAggregation create(
+  public static DistributionAggregation create(
       long count, double mean, double sum, Range range, List<Tag> tags, List<Long> bucketCounts) {
-    return new DistributionAggregation(count, mean, sum, range, tags,
+    return createInternal(count, mean, sum, range, tags,
         Collections.unmodifiableList(new ArrayList<Long>(bucketCounts)));
   }
 
-  /**
-   * {@link Tag}s associated with this {@link DistributionAggregation}.
-   *
-   * <p>Note: The returned list is unmodifiable, attempts to update it will throw an
-   * UnsupportedOperationException.
-   */
-  public final List<Tag> getTags() {
-    return tags;
+  private static DistributionAggregation createInternal(
+      long count, double mean, double sum, Range range, List<Tag> tags, List<Long> bucketCounts) {
+    return new AutoValue_DistributionAggregation(count, mean, sum, range, tags, bucketCounts);
   }
 
   /**
    * The number of values in the population. Must be non-negative.
    */
-  public long getCount() {
-    return count;
-  }
+  public abstract long getCount();
 
   /**
    * The arithmetic mean of the values in the population. If {@link #getCount()} is zero then this
    * value must also be zero.
    */
-  public double getMean() {
-    return mean;
-  }
+  public abstract double getMean();
 
   /**
    * The sum of the values in the population.  If {@link #getCount()} is zero then this values must
    * also be zero.
    */
-  public double getSum() {
-    return sum;
-  }
+  public abstract double getSum();
 
   /**
    * The range of the population values. If {@link #getCount()} is zero then this returned range is
    * implementation-dependent.
    */
-  public Range getRange() {
-    return range;
-  }
+  public abstract Range getRange();
```

Comment:
Should we comment something like:
"Returns the range ..."

Also probably better to copy the same in "@return ...", even if it repeats the same thing the javadoc looks better.